x-pack/plugins/profiling/public/components/flamegraph.tsx

```diff
@@ -5,11 +5,76 @@
  * 2.0.
  */
 
+import { Chart, ColumnarViewModel, Datum, Flame, PartialTheme, Settings } from '@elastic/charts';
+import { EuiFlexGroup, EuiFlexItem, EuiPanel } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
 import React, { useContext, useMemo } from 'react';
+import { asPercentage } from '../utils/formatters/as_percentage';
+import { FlameGraphContext } from './contexts/flamegraph';
 
-import { Chart, ColumnarViewModel, Datum, Flame, PartialTheme, Settings } from '@elastic/charts';
+function TooltipRow({ value, label }: { value: string; label: string }) {
+  return (
+    <EuiFlexItem>
+      <EuiFlexGroup direction="row" gutterSize="xs">
+        <EuiFlexItem grow={false} style={{ fontWeight: 'bold' }}>
+          {label}
+        </EuiFlexItem>
+        <EuiFlexItem>{value}</EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiFlexItem>
+  );
+}
 
-import { FlameGraphContext } from './contexts/flamegraph';
+function FlameGraphTooltip({
+  label,
+  countInclusive,
+  countExclusive,
+  samples,
+  totalSamples,
+}: {
+  samples: number;
+  label: string;
+  countInclusive: number;
+  countExclusive: number;
+  totalSamples: number;
+}) {
+  return (
+    <EuiPanel>
+      <EuiFlexGroup direction="column" gutterSize="m">
+        <EuiFlexItem
+          style={{
+            width: 400,
+            overflowWrap: 'anywhere',
+          }}
+        >
+          {label}
+        </EuiFlexItem>
+        <EuiFlexItem style={{ whiteSpace: 'nowrap' }}>
+          <EuiFlexGroup direction="column" gutterSize="xs">
+            <TooltipRow
+              label={i18n.translate('xpack.profiling.flameGraphTooltip.inclusiveCpuLabel', {
+                defaultMessage: `Inclusive CPU:`,
+              })}
+              value={asPercentage(countInclusive / totalSamples, 1)}
```

Comment:
Can we have 2 decimal places ?
Or something adaptive like "2 relevant digits", e.g. 26%, 3.5%, 0.17%, 0.073% etc ? (I am also ok with 3 relevant digits here.)

Reply:
done 👍